packages/react-components/src/components/chart/events/useHandleChartEvents.ts

```diff
@@ -0,0 +1,113 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { EChartsOption, EChartsType } from 'echarts';
+
+export const useHandleChartEvents = (chartRef: MutableRefObject<EChartsType | null>) => {
+  const [optionState, setOptionState] = useState<EChartsOption>({});
+
+  // When dragging on the chart, turn off tooltip and set cursor to grabbing
+  const setOnDragStart = (chart: EChartsType | null) => {
+    setOptionState({
+      tooltip: {
+        show: false,
+      },
+    });
+    chart?.getZr().setCursorStyle('grabbing');
+  };
+
+  // When dragging is done turn back on the tooltip and set cursor to grab
+  // Use when the shift key is still held down but mouse is up
+  const setOnDragEnd = (chart: EChartsType | null) => {
+    setOptionState({
+      tooltip: {
+        show: true,
+      },
+    });
+    chart?.getZr().setCursorStyle('grab');
+  };
+
+  // Set up event listeners for pressing shift and dragging the mouse to update
+  // the cursor and tooltip
+  useEffect(() => {
+    const chart = chartRef?.current;
+
+    let shiftDown = false;
+    let mouseDown = false;
+    let prevIsGrabbing = false;
+
+    // When shift is held down show the grab cursor by default
+    // When mouse is also held down it is a grabbing event
+    const shiftDownListener = (event: KeyboardEvent) => {
+      shiftDown = event.shiftKey;
+      if (mouseDown && shiftDown) {
+        setOnDragStart(chart);
+      } else if (shiftDown) {
+        chart?.getZr().setCursorStyle('grab');
+      }
+    };
+
+    // When shift is released go back to the default cursor
+    const shiftUpListener = (event: KeyboardEvent) => {
+      if (event.key.toLowerCase() === 'shift') {
+        shiftDown = false;
+        chart?.getZr().setCursorStyle('default');
+      }
+    };
+
+    window.addEventListener('keydown', shiftDownListener);
```

Comment:
curious : the event handler should be on the chart right ? why on the entire window ?

Reply:
echarts does not have an event interface for `keydown` or `keyup` so I needed to define my own.

Reply:
curious: could we leverage HotKeys which is already integrated ? my concern is that this handler will get triggered anytime the user presses shift ... i.e it is not specific to charts ,,

Reply:
Can you give me more info about hot keys? This handler will get triggered on shift, but it only impacts the xyplot chart. If you have multiple of those charts on your dashboard then hovering around with shift will set the `grab` cursor style on those charts. This doesn't impact anything else.

Reply:
https://github.com/awslabs/iot-app-kit/blob/main/packages/react-components/src/components/chart/baseChart.tsx#L153 --> take a look at this, its probably better to use this instead of a global event handler ? 

this is the library https://www.npmjs.com/package/react-hotkeys

Reply:
Updated to use hotkeys after discussing offline